### PR TITLE
Arnej/rank filter in struct field

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst please review
@jobergum FYI

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/searchdefinition/derived/AttributeFields.java [hunk 1]
```diff
@@ -134,6 +134,11 @@ private void deriveAttributeAsArrayType(ImmutableSDField field) {
         }
         Attribute attribute = field.getAttributes().get(field.getName());
         if (attribute != null) {
+            Ranking ranking = field.getRanking();
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/searchdefinition/derived/AttributeFields.java:137]: I suggest creating a helper function for this (e.g. applyRankingSettings() / considerApplyRankingSettings()) and use that in deriveAttribute() as well to ensure we handle this the same way in both code paths.

**Review by geirst — APPROVED** [review]
👍

**Comment by arnej27959** [comment]
note: this only works with the new SD parser, we don't know for sure when that is enabled everywhere.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**